lib/Sema/CSSimplify.cpp

```diff
@@ -2287,6 +2279,15 @@ bool ConstraintSystem::repairFailures(
             rhs)) {
       conversionsOrFixes.push_back(fix);
     }
+
+    auto elementTy = lhs->getArrayElementType();
+    if (elt.getKind() == ConstraintLocator::ApplyArgToParam &&
+        elementTy && elt.getParameterFlags().isVariadic()) {
```

Comment:
You could use `ConstraintSystem::isArrayType` here instead, which saves you from having to introduce `isArrayDecl` & `getArrayElementType`.

Reply:
Also you no longer need to check for `elt.getKind() == ConstraintLocator::ApplyArgToParam` now, as there's an early exit for all other cases a few lines above.